src/SW_Main.c

```
@@ -68,63 +75,93 @@ int main(int argc, char **argv) {
+                                sw_template.Model.cum_monthdays,
+								sw_template.Model.days_in_month, &LogInfo);
     if(LogInfo.stopRun) {
         goto finishProgram;
     }
 
 	// initialize simulation run (based on user inputs)
-	SW_CTL_init_run(&sw, &LogInfo);
+	SW_CTL_init_run(&sw_template, &LogInfo);
     if(LogInfo.stopRun) {
         goto finishProgram;
     }
 
   // initialize output
-	SW_OUT_set_ncol(sw.Site.n_layers, sw.Site.n_evap_lyrs, sw.VegEstab.count,
-					sw.GenOutput.ncol_OUT);
-	SW_OUT_set_colnames(sw.Site.n_layers, sw.VegEstab.parms,
-						sw.GenOutput.ncol_OUT, sw.GenOutput.colnames_OUT,
+	SW_OUT_set_ncol(sw_template.Site.n_layers, sw_template.Site.n_evap_lyrs,
+                    sw_template.VegEstab.count, sw_template.GenOutput.ncol_OUT);
+	SW_OUT_set_colnames(sw_template.Site.n_layers, sw_template.VegEstab.parms,
+						sw_template.GenOutput.ncol_OUT, sw_template.GenOutput.colnames_OUT,
 						&LogInfo);
     if(LogInfo.stopRun) {
         goto finishProgram;
     }
-	SW_OUT_create_files(&sw.FileStatus, sw.Output, sw.Site.n_layers,
-	                    PathInfo.InFiles, &sw.GenOutput, &LogInfo); // only used with SOILWAT2
+	SW_OUT_create_files(&sw_template.FileStatus, sw_template.Output, sw_template.Site.n_layers,
+	                    PathInfo.InFiles, &sw_template.GenOutput, &LogInfo); // only used with SOILWAT2
     if(LogInfo.stopRun) {
         goto closeFiles;
     }
 
 	if(EchoInits) {
-		_echo_all_inputs(&sw);
+		_echo_all_inputs(&sw_template);
 	}
 
-  // run simulation: loop through each year
-	SW_CTL_main(&sw, SW_OutputPtrs, &LogInfo);
+  // run simulations: loop over simulation set
+    for(suid = startSimSet; suid < endSimSet; suid++)
+    {
+        ncStartSuid = SW_DOM_calc_ncStartSuid(&SW_Domain, suid);
+
+        if(SW_DOM_CheckProgress(SW_Domain.DomainType, ncStartSuid)) {
+            SW_CTL_run_sw(&sw_template, &SW_Domain, ncStartSuid, NULL,

```

The domain loop currently fails silently if the domain is larger than one